Remora.Discord.Commands/Conditions/Attributes/RequireContextAttribute.cs

```diff
@@ -32,17 +33,33 @@ namespace Remora.Discord.Commands.Conditions
     public class RequireContextAttribute : ConditionAttribute
     {
         /// <summary>
-        /// Gets the command context.
+        /// Gets the command grouped contexts.
         /// </summary>
-        public ChannelContext Context { get; }
+        public ChannelContext[]? ChannelContexts { get; }
+
+        /// <summary>
+        /// Gets the command individual channel type contexts.
+        /// </summary>
+        public ChannelType[]? ChannelTypes { get; }
```

Comment:
You can remove one of these arrays and make it non-nullable by converting the constructor parameters to the appropriate types. Personally, I think it'd be most appropriate to remove `ChannelContexts` and convert each grouped context into its constituent parts. Saves on logic and lets users have a simpler API, while at the same time allowing greater detailed control :)

Reply:
Removing the ChannelContexts will be a breaking change, however i have no issue on doing that but i didn't understand that part:
> I think it'd be most appropriate to remove ChannelContexts and **convert each grouped context into its constituent parts**.